lib/ActivityStreams.js

```diff
@@ -75,6 +80,27 @@ ActivityStreams.prototype = {
   },
 
   /**
+   * Every 6 hours pro-actively prime our emebedly cache with a normal content page load
+   */
+  _primeCache() {
+    setTimeout(() => {
+      PlacesProvider.links.getTopFrecentSites.then(links => {
+        PreviewProvider.preview.primeCache(links);
+      });
+      PlacesProvider.links.getRecentBookmarks.then(links => {
+        PreviewProvider.preview.primeCache(links);
+      });
+      PlacesProvider.links.getRecentLinks.then(links => {
+        PreviewProvider.preview.primeCache(links);
+      });
+      PlacesProvider.links.getFrecentLinks.then(links => {
+        PreviewProvider.preview.primeCache(links);
+      });
+      this._primeCache();
+    }, PRIME_CACHE_TIMEOUT);
```

Comment:
If it does what I think it does, when you call `setTimeout`, it won't execute the callback you provide immediately.

i.e. if you run:
```javascript
let start = (new Date()).getTime();
setTimeout(() => {
  let step = (new Date()).getTime();
  console.log(`elapsed: ${step - start}`);
}, 3000);
```

The result will be close to 3000, meaning that the callback doesn't execute when you set the timeout. In this case, it will only execute after `PRIME_CACHE_TIMEOUT` milliseconds.

If I understand correctly, your intention is to populate the cache at bootup time, correct?